### PR TITLE
[route] Ignore port when picking host

### PR DESCRIPTION
**Pull request by stevenzzzz**
…ned in host header during host matching.

Signed-off-by: Xin Zhuang <stevenzzz@google.com>

Commit Message: Add an option in RouteConfiguration that when enabled, ignore any port number in host header during host matching.

Additional Description:
Risk Level: LOW (no behavior change with default value).
Testing: unit test
Release Notes: Add an option in RouteConfiguration that when enabled, ignore any port number in host header during host matching.
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/21272

### CODE REVIEW

File: api/envoy/config/route/v3/route.proto [hunk 1]
```diff
@@ -129,6 +129,11 @@ message RouteConfiguration {
   // Specify a set of default request mirroring policies which apply to all routes under its virtual hosts.
   // Note that policies are not merged, the most specific non-empty one becomes the mirror policies.
   repeated RouteAction.RequestMirrorPolicy request_mirror_policies = 13;
+
+  // By default, port in :authority header (if any) is used in host matching.
```
[the remark this answers is not in the document]
Comment (stevenzzzz) [reply api/envoy/config/route/v3/route.proto:133]: yeah, the port number in config(if any) is not mentioned. updated the doc

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**